library/std/src/io/entropy.rs

```diff
@@ -0,0 +1,125 @@
+use super::{BorrowedBuf, BorrowedCursor, Read, Result};
+use crate::sys::entropy as sys;
+
+/// A reader which returns random bytes from the system entropy source.
+///
+/// This struct is generally created by calling [`entropy()`]. Please
+/// see the documentation of [`entropy()`] for more details.
+#[derive(Debug)]
+#[unstable(feature = "io_entropy", issue = "none")]
+pub struct Entropy {
+    insecure: bool,
+}
+
+impl Entropy {
+    pub(crate) fn set_insecure(&mut self, insecure: bool) {
+        self.insecure = insecure;
+    }
+}
+
+#[unstable(feature = "io_entropy", issue = "none")]
+impl Read for Entropy {
+    #[inline]
+    fn read(&mut self, buf: &mut [u8]) -> Result<usize> {
+        sys::Entropy { insecure: self.insecure }.read(buf)
+    }
+
+    #[inline]
+    fn read_exact(&mut self, buf: &mut [u8]) -> Result<()> {
+        let mut buf = BorrowedBuf::from(buf);
+        self.read_buf_exact(buf.unfilled())
+    }
+
+    #[inline]
+    fn read_buf(&mut self, buf: BorrowedCursor<'_>) -> Result<()> {
+        sys::Entropy { insecure: self.insecure }.read_buf(buf)
+    }
+
+    #[inline]
+    fn read_buf_exact(&mut self, buf: BorrowedCursor<'_>) -> Result<()> {
+        sys::Entropy { insecure: self.insecure }.read_buf_exact(buf)
+    }
+}
+
+/// Constructs a new handle to the system entropy source.
+///
+/// Reads from the resulting reader will return high-quality random data that
+/// is suited for cryptographic purposes (by the standard of the platform defaults).
+///
+/// Be aware that, because the data is of very high quality, reading high amounts
+/// of data can be very slow, and potentially slow down other processes requiring
+/// random data. Use a pseudo-random number generator if speed is important.
```

Comment:
Since we do this elsewhere in libstd with other crates, perhaps we could just straight-up recommend the `rand` crate for this purpose? Especially since this is an org-maintained crate.

Reply:
We have been trying to not advertise external crates from std docs. And there are other potential options beside `rand` even if they're more niche.